From e9b94787a8c952a61b97f0871484f2ed8dd493a7 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:15 -0400
Subject: [powerpc] pseries: CONFIG_PSERIES_MSI should depend on PPC_PSERIES

Message-id: <1410545655-205645-207-git-send-email-dzickus@redhat.com>
Patchwork-id: 94313
O-Subject: [RHEL7 PATCH 206/626] powerpc/pseries: CONFIG_PSERIES_MSI should depend on PPC_PSERIES
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 1edb55a473eb1151f54590371761fdeb35b3b593
Author: Michael Ellerman <michael@ellerman.id.au>
Date:   Thu Sep 26 17:52:16 2013 +1000

    powerpc/pseries: CONFIG_PSERIES_MSI should depend on PPC_PSERIES

    Previously PSERIES_MSI depended on PPC_PSERIES via EEH. However in
    commit 317f06d "powerpc/eeh: Move common part to kernel directory" we
    made CONFIG_EEH selectable on POWERNV. That leaves us with PSERIES_MSI
    being live even when PSERIES=n. Fix it by making PSERIES_MSI depend
    directly on PPC_PSERIES.

    Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
    Reviewed-by: Gavin Shan <shangw@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/pseries/Kconfig b/arch/powerpc/platforms/pseries/Kconfig
index 3763418..b54e933 100644
--- a/arch/powerpc/platforms/pseries/Kconfig
+++ b/arch/powerpc/platforms/pseries/Kconfig
@@ -34,7 +34,7 @@ config PPC_SPLPAR
 
 config PSERIES_MSI
        bool
-       depends on PCI_MSI && EEH
+       depends on PCI_MSI && PPC_PSERIES && EEH
        default y
 
 config PSERIES_ENERGY
-- 
1.7.1